Readme and notebook on using Zarr on CESNET Swift (#15)
pangeo-data/pangeo-eosc: EGI-CLI-Swift.md

# Using fedcloud and openstack slients

## First install required Python packages

```

> conda create -n egi python

Reviewer: `jq` will be useful when parsing JSON output from `fedcloudclient`
```suggestion
conda create -n egi python jq --yes
conda activate egi
```

> pip install openstackclient

Reviewer: `pip install fedcloudclient` will install this as a dependency.
```suggestion
```

pip install fedcloudclient
```

> 

Reviewer: ```suggestion
See more information about `fedcloudclient` in: https://fedcloudclient.fedcloud.eu/

```

You should be able to issue:
```
fedcloud token check
```

But you don't have a token yet.

> ## Connect to EGI

Reviewer: ```suggestion
## Get your access token from EGI Check-In
```

> Go to https://aai.egi.eu/token/ to optain your client token. Copy your access token, and then set it:

Reviewer: ```suggestion
Go to https://aai.egi.eu/token/ to [obtain your access token](https://docs.egi.eu/users/aai/check-in/obtaining-tokens/token-portal/). Copy your access token, and then set it:
```

```
export OIDC_ACCESS_TOKEN=<your_token>
fedcloud token check
```

Last command should return a valid token.

## Use Openstack

> You need to set other env variables to simplify things:
> 
> ```
> export EGI_SITE=CESNET-MCC
> export EGI_VO=vo.pangeo.eu
> ```

Reviewer: This is about to change in an upcoming release of `fedcloudclient` so I would rather leave the CLI options below.
```suggestion
```

Then the following command should work:
```

> fedcloud openstack container list

Reviewer: ```suggestion
fedcloud openstack --site CESNET-MCC --vo vo.pangeo.eu container list
```

```

> ## Retrieve Openstack swift important values

Reviewer: ```suggestion
## Retrieve Openstack swift credentials
```

```

> # explore sites with swift storage
> $ fedcloud endpoint list --service-type org.openstack.swift --site ALL_SITES
> 

Reviewer: This can be skipped:
```suggestion

```

# get OS_AUTH_URL
$ fedcloud openstack --site CESNET-MCC --vo vo.pangeo.eu catalog show keystone

# get OS_AUTH_TOKEN
$ fedcloud openstack --site CESNET-MCC --vo vo.pangeo.eu token issue \
  -c id \
  -f value

# get OS_STORAGE_URL for your site and Virtual Organisation
$ fedcloud openstack --site CESNET-MCC --vo vo.pangeo.eu catalog show swift
```

> You'll need OS_AUTH_TOKEN and OS_STORAGE_URL in order to interact with Swift using Zarr.

Reviewer: You'll need OS_AUTH_TOKEN and OS_STORAGE_URL in order to interact with Swift using Zarr.

## Retrieve S3 credentials

CESNET provides the following self-service to get S3 credentials:
https://docs.cloud.muni.cz/cloud/advanced-features/#s3-credentials

Using `fedcloudclient` you can do:
```
fedcloud openstack --site CESNET-MCC --vo vo.pangeo.eu ec2 credentials create 
fedcloud openstack --site CESNET-MCC --vo vo.pangeo.eu ec2 credentials list
```

This will provide `access` and `secret` keys. The `endpoint` URL is: https://object-store.cloud.muni.cz/
